controllers/toolchaincluster_resources/toolchaincluster_resources_controller_test.go

```diff
@@ -0,0 +1,91 @@
+package toolchaincluster_resources
+
+import (
+	"context"
+	"embed"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+//go:embed testdata/cluster-role.yaml
+var clusterRoleFS embed.FS
+
+//go:embed testdata/service-account.yaml
+var serviceAccountFS embed.FS
+
+func TestToolchainClusterResources(t *testing.T) {
+	// given
+	status := test.NewClusterStatus(toolchainv1alpha1.ToolchainClusterReady, v1.ConditionTrue)
+	toolchainCluster, sec := test.NewToolchainCluster("east", test.MemberOperatorNs, "secret", status, nil)
+	toolchainCluster.Spec.CABundle = "ZHVtbXk="
+	cl := test.NewFakeClient(t, toolchainCluster, sec)
+	service := cluster.NewToolchainClusterServiceWithClient(cl, logf.Log, test.MemberOperatorNs, 3*time.Second, func(config *rest.Config, options client.Options) (client.Client, error) {
+		return client.New(config, options)
+	})
+	defer service.DeleteToolchainCluster("east")
```

Comment:
why do we need all these toolchainCluster stuff? it looks like a copy-paste.
A simple empty fake client should be enough:
```suggestion
	cl := test.NewFakeClient(t)
```